### ISSUE TITLE
Full list of TimeZoneInfos

### ISSUE THREAD

**Issue by cocowalla** [mattjohnsonpint/TimeZoneNames#49]
I'm sure I'm missing something really obvious here, but is there a way to get a list of `TimeZoneInfo`s for _all_ canonical IANA timezone IDs that are valid as of a given date?

**Comment by mj1856** [comment]
Not sure exactly what you mean exactly.  Can you provide an example of what you'd expect?

[TimeZoneConverter ](https://github.com/mj1856/TimeZoneConverter) has a `TZConvert.KnownWindowsTimeZoneIds` collection, if that's what you mean.  Though it is always the current fully-up-to-date list.  I don't track history of when the different time zones were made available (yet).

**Comment by cocowalla** [comment]
What I'm looking for is `TZConvert.KnownWindowsTimeZoneIds`, but:

1. Filtered to only include _canonical_ time zones and
2. Filtered to only include timezones valid from a specified threshold (that is, they are not deprecated)

**Comment by mj1856** [comment]
I assume you are referencing the format column in [this Wikipedia page](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones)?  One thing to keep in mind is that "deprecated" zones there really just mean that they aren't in the prefered area/locality format, so they should be generally removed from a pick-list.  I assume that is your use case?

But that doesn't mean they don't work any more.  In general, once a time zone id is introduced, it doesn't tend to go away.  There has been one exception (`Canada/East-Saskatchewan` [discussed here](https://github.com/mj1856/TimeZoneConverter/issues/12#issuecomment-368645081)), but I know of no others.

Also, I don't this status in these libraries presently.  I will consider it for the future and leave this issue open to remind me, but I make no promises.  Thanks.

**Comment by cocowalla** [comment]
Yes, this is basically for allowing users to select their timezone from a dropdown, and it seems prudent to prune the (rather large) list by removing ones that are no longer in use. Since `GetTimeZonesForCountry` has a threshold parameter, I just assumed there was some simple way to do this for *all* timezones... other than calling `GetLanguageCodes` and feeding the result into `GetCountryNames`, then feeding the result of *that* into `GetTimeZonesForCountry` :)

**Comment by mj1856** [comment]
Oh, I'm sorry - I misunderstood. I see what you mean now.

Yes, today you can build this list like so:

var threshold = DateTimeOffset.Now;

```csharp
var ids = TZNames.GetCountryNames("en").Keys
    .SelectMany(code => TZNames.GetTimeZoneIdsForCountry(code, threshold))
    .Concat(TZNames.GetFixedTimeZoneIds())
    .Distinct()
    .OrderBy(x => x);
```

And yes, I agree this should be built in as a simpler single function.  Thanks for the suggestion.
